module/VuFind/src/VuFind/ILS/Connection.php

```diff
@@ -1325,6 +1328,89 @@ public function callIlsWithFailover($methodName, $params)
         );
     }
 
+    /**
+     * Get timed blocks for a method from driver configuration
+     *
+     * @param string $methodName Method to check
+     *
+     * @return array Array with keys 'start', 'end', 'recurring'
+     *               or empty array if no blocks are found
+     */
+    public function getMethodTimedBlocks(string $methodName): array
+    {
+        $functionConfig = $this->checkCapability('getConfig', ['TimedBlocks'])
+            ? $this->getDriver()->getConfig('TimedBlocks')
+            : [];
+
+        if (!isset($functionConfig[$methodName])) {
+            return [];
+        }
+        $methodBlock = $functionConfig[$methodName];
+        $startDate = strtotime($methodBlock['startDate'] ?? '');
+        $endDate = $methodBlock['endDate'] ?? '';
+        $noEndHours = empty(explode(' ', $endDate, 2)[1]);
+        if ($endDate && $noEndHours) {
+            $endDate .= ' 23.59.59';
```

Comment:
If we recommend HH:MM in the config, should this be ` 23:59.59` for consistency?